core/src/test/java/org/apache/calcite/test/MaterializedViewRelOptRulesTest.java

```diff
@@ -338,6 +341,104 @@ protected final MaterializedViewFixture sql(String materialize,
         .ok();
   }
 
```

Comment:
Could you add one or more tests explicitly covering [your claim](https://issues.apache.org/jira/browse/CALCITE-5240?focusedCommentId=17608900&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17608900
) that the behaviour is unchanged when we have more complex cases (a predicate already exists, we have a predicate in a join, etc.)?

Reply:
I have added `testAggregateViewWithPredicate` to test when a predicate already exists in the view and `testJoinViewWithoutPredicate`, `testJoinViewWithPredicateSameAsQueryPredicate`, `testJoinViewWithPredicateDifferentThanQueryPredicate` to test views that have a join with predicate.